app/scenes/Document/components/Document.js

```diff
@@ -54,157 +50,181 @@ Are you sure you want to discard them?
 `;
 
 type Props = {
-  match: Match,
-  history: RouterHistory,
-  location: LocationWithState,
   abilities: Object,
   document: Document,
   revision: Revision,
   readOnly: boolean,
   onCreateLink: (title: string) => Promise<string>,
   onSearchLink: (term: string) => any,
   theme: Theme,
-  auth: AuthStore,
-  ui: UiStore,
 };
 
-@observer
-class DocumentScene extends React.Component<Props> {
-  @observable editor = React.createRef();
-  @observable isUploading: boolean = false;
-  @observable isSaving: boolean = false;
-  @observable isPublishing: boolean = false;
-  @observable isDirty: boolean = false;
-  @observable isEmpty: boolean = true;
-  @observable moveModalOpen: boolean = false;
-  @observable lastRevision: number = this.props.document.revision;
-  @observable title: string = this.props.document.title;
-  getEditorText: () => string = () => this.props.document.text;
```

Comment:
`getEditorText` was used in a cryptic way. I was confused about whether it is a function or a variable. So, I rewrote it as a state.

Reply:
`getEditorText` is a function that returns the editor text as markdown, the reason it's a function is because converting the entire document state to markdown can be a relatively expensive operation so we only want to do it when neccessary, not every time the doc changes.

It might want to be `setGetEditorText`, lol.